packages/git/src/browser/blame/blame-manager.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (C) 2018 TypeFox and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License'); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { inject, injectable } from 'inversify';
+import { Git, GitFileBlame } from '../../common';
+import { GitRepositoryTracker } from '../git-repository-tracker';
+import URI from '@theia/core/lib/common/uri';
+
+@injectable()
+export class BlameManager {
+
+    @inject(Git)
+    protected readonly git: Git;
+
+    @inject(GitRepositoryTracker)
+    protected readonly repositoryTracker: GitRepositoryTracker;
+
+    isBlameable(uri: string): boolean {
+        return !!this.repositoryTracker.getPath(new URI(uri));
```

Comment:
I just tested and it works nicely with files with multiple commits :)

I'm just not sure about the isBlameable check, it seems to check if the editor uri is part of a git repo am I right? I tried showing git annotations for a new file which wasn't part of the index of the git repo and I got this error:

```
messages.js:46 Uncaught (in promise) Error: Request blame failed with message: fatal: no such path 'git2/test3' in HEAD
```
I think we should check to see if that file is part the index maybe? Or actually show an error to the user. Calling `git blame git2/test3` returns `fatal: no such path 'test3' in HEAD`

Reply:
thanks for testing! 👍 
It used to work previously, need to debug and/or check recent changes.

Reply:
I checked with codelense, it also shows no annotations for uncommitted files.

Reply:
I'll align with that behavior, as it makes things a bit easier. currently senseless annotations are shown for new & staged files.

Reply:
fixed.